flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/FlinkResourceManager.java

```diff
@@ -0,0 +1,796 @@
+	 *
+	 * @param registered The list of TaskManagers that the JobManager knows.
+	 * @return The subset of TaskManagers that the resource manager can confirm to be alive.
+	 */
+	protected abstract Collection<WorkerType> reacceptRegisteredWorkers(Collection<ResourceID> registered);
+
+	/**
+	 * Gets the number of requested workers that have not yet been granted.
+	 *
+	 * @return The number pending worker requests. Must never be smaller than 0.
+	 */
+	protected abstract int getNumWorkerRequestsPending();
+
+	/**
+	 * Gets the number of containers that have been started, but where the TaskManager
+	 * has not yet registered at the job manager.
+	 *
+	 * @return The number of started containers pending TaskManager registration.
+	 * Must never be smaller than 0.
+	 */
+	protected abstract int getNumWorkersPendingRegistration();
+
+	// ------------------------------------------------------------------------
+	//  Info messaging
+	// ------------------------------------------------------------------------
+
+	protected void sendInfoMessage(String message) {
+		for (ActorRef listener : infoMessageListeners) {
+			listener.tell(decorateMessage(new InfoMessage(message)), self());
+		}
+	}
+
+
+	// ------------------------------------------------------------------------
+	//  Startup
+	// ------------------------------------------------------------------------
+
+	public static void startResourceManager() {
+
+	}
+
+	/**
+	 * Starts the resource manager actors.
+	 * @param configuration The configuration for the resource manager
+	 * @param actorSystem The actor system to start the resource manager in
+	 * @param leaderRetriever The leader retriever service to intialize the resource manager
+	 * @param resourceManagerClass The class of the ResourceManager to be started
+	 * @return ActorRef of the resource manager
+	 */
+	public static ActorRef startResourceManagerActors(
+		Configuration configuration,
+		ActorSystem actorSystem,
+		LeaderRetrievalService leaderRetriever,
+		Class<? extends FlinkResourceManager<?>> resourceManagerClass
+	) {
+		return startResourceManagerActors(
+			configuration, actorSystem, leaderRetriever, resourceManagerClass,
+			RESOURCE_MANAGER_NAME + "-" + UUID.randomUUID());
+	}
+
+	/**
+	 * Starts the resource manager actors.
+	 * @param configuration The configuration for the resource manager
+	 * @param actorSystem The actor system to start the resource manager in
+	 * @param leaderRetriever The leader retriever service to intialize the resource manager
+	 * @param resourceManagerClass The class of the ResourceManager to be started
+	 * @param resourceManagerActorName The name of the resource manager actor.
+	 * @return ActorRef of the resource manager
+	 */
+	public static ActorRef startResourceManagerActors(
+		Configuration configuration,
+		ActorSystem actorSystem,
+		LeaderRetrievalService leaderRetriever,
+		Class<? extends FlinkResourceManager<?>> resourceManagerClass,
+		String resourceManagerActorName
+	) {
```

Comment:
I just don't feel so strong about it but I can see how it can drive other people nuts. I'll do another pass to fix some of these style inconsistencies.